commit 8064d12129c92209ecd08461adf41918b21263e4
Author: Scott Moser <scott.moser@chainguard.dev>
Date:   Fri Sep 6 19:44:49 2024 +0000

    Fix building on gcc-14.
    
    I'm able to get this to build on gcc-14 with
    this change and CXX_FLAGS containing all of the following:
    
        -Wno-error=maybe-uninitialized
        -Wno-error=conversion
        -Wno-error=float-conversion
        -Wno-error=format-nonliteral

diff --git a/src/parser/wat-parser.h b/src/parser/wat-parser.h
index 041ba1d58..0bfb829cb 100644
--- a/src/parser/wat-parser.h
+++ b/src/parser/wat-parser.h
@@ -34,12 +34,17 @@ Result<> parseModule(Module& wasm, Lexer& lexer);
 
 Result<Literal> parseConst(Lexer& lexer);
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+
 struct InvokeAction {
   std::optional<Name> base;
   Name name;
   Literals args;
 };
 
+#pragma GCC diagnostic pop
+
 struct GetAction {
   std::optional<Name> base;
   Name name;
